mathlibtools/lib.py

```diff
@@ -639,6 +639,38 @@ def import_graph(self) -> 'ImportGraph':
         self._import_graph = G
         return G
 
+    def reduce_imports(self, file: str) -> Iterable[List[Tuple[str, str]]]:
+        """
+        An iterator over files with removable imports, for each file yielding
+        a list of removable imports in the format
+        `("removable.import", "source.file")`.
+        """
+        # Importing networkx slow, so don't do it until this function
+        # is called.
+        import networkx as nx # type: ignore
+        G = self.import_graph
+        if file:
+            G = G.ancestors(file)
+        H = nx.transitive_reduction(G)
+        if file:
+            fs = [file]
+        else:
+            fs = G.nodes
+        for f in fs:
+            if f == "all":
+                continue
+            Gf = [e for e in G.edges if e[1] == f]
+            Hf = [e for e in H.edges if e[1] == f]
```

Comment:
`Hf` is unused?

Reply:
Thanks, this should have been referenced in the next line.